db/compaction_picker.cc

```diff
@@ -1275,6 +1294,41 @@ bool LevelCompactionPicker::PickCompactionBySize(VersionStorageInfo* vstorage,
   return inputs->size() > 0;
 }
 
+bool LevelCompactionPicker::PickIntraL0Compaction(
+    VersionStorageInfo* vstorage, CompactionInputFiles* inputs) {
+  inputs->clear();
+  const std::vector<FileMetaData*>& level_files =
+      vstorage->LevelFiles(0 /* level */);
+  if (level_files.size() == 0 || level_files[0]->being_compacted) {
```

Comment:
How about we take whichever is larger:
1. current L0 score
2. current L0 size / base_level_size_base.

Because we know have L0->L0 compaction, the size indicated by 1 is now exploded. I think the number can be balanced with 2.